modules/dmrpp_module/DMZ.cc

```diff
@@ -2020,38 +2025,54 @@ bool DMZ::process_chunks(BaseType *btp, const xml_node &var_node) const
     // Look for the chunksDimensionSizes element - it will not be present for contiguous data
     process_cds_node(dc(btp), chunks);
 
-    // Chunks for this node will be held in the var_node siblings.
+    // If child node "dmrpp:chunk" is found, the child node "dmrpp:block" will be not present.
+    // They are mutual exclusive. 
+
+    bool is_chunked_storage = false;
     for (auto chunk = chunks.child("dmrpp:chunk"); chunk; chunk = chunk.next_sibling()) {
         if (is_eq(chunk.name(), "dmrpp:chunk")) {
-            process_chunk(dc(btp), chunk);
+            is_chunked_storage = true;
+            break;
         }
     }
 
-    // Blocks for this node, we need to first check if there is only one block. If this is the case,
-    // we should issue an error.
-    unsigned int block_count = 0;
-    for (auto chunk = chunks.child("dmrpp:block"); chunk; chunk = chunk.next_sibling()) {
-        if (is_eq(chunk.name(), "dmrpp:block")) {
-            block_count++;
+    if (is_chunked_storage) {
+        // Chunks for this node will be held in the var_node siblings.
+        for (auto chunk = chunks.child("dmrpp:chunk"); chunk; chunk = chunk.next_sibling()) {
+            if (is_eq(chunk.name(), "dmrpp:chunk")) {
+                process_chunk(dc(btp), chunk);
+            }
         }
-        if (block_count >1)
-            break;
     }
-    if (block_count == 1)
-        throw BESInternalError(" The number of linked block is 1, but it should be > 1.", __FILE__, __LINE__);
-    if (block_count >1) {
-        // set using linked block
-        dc(btp)->set_using_linked_block();
-        // reset the count to 0 to process the blocks.
-        block_count = 0;
+
+    else {
+
+        // Blocks for this node, we need to first check if there is only one block. If this is the case,
+        // we should issue an error.
+        unsigned int block_count = 0;
         for (auto chunk = chunks.child("dmrpp:block"); chunk; chunk = chunk.next_sibling()) {
             if (is_eq(chunk.name(), "dmrpp:block")) {
-                process_block(dc(btp), chunk, block_count);
-                BESDEBUG(PARSER, prolog << "This count of linked block of this variable is: " << block_count << endl);
                 block_count++;
             }
+            if (block_count >1)
+                break;
+        }
+        if (block_count == 1)
+            throw BESInternalError(" The number of linked block is 1, but it should be > 1.", __FILE__, __LINE__);
+        if (block_count >1) {
+            // set using linked block
+            dc(btp)->set_using_linked_block();
+            // reset the count to 0 to process the blocks.
+            block_count = 0;
+            for (auto chunk = chunks.child("dmrpp:block"); chunk; chunk = chunk.next_sibling()) {
```

Comment:
Should/could 'chunk' in the above be a reference?

Reply:
I did this by following the existing code. So I think it must have a reason not to use the reference. I will try to remember to check the use of reference in my next pull request.